src/main/java/com/google/sps/servlets/LocationsServlet.java

```diff
@@ -60,27 +65,49 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         QueryHelper.getDistanceBasedEntities(
             EntityType.LISTING, numListings, selectedOffice, distanceInKilometers);
 
-    List<Entity> filteredEntityListByLatitude =
-        CoordinateCalculator.filterOutOfRangeLatitudeEntities(
-            distanceInKilometers,
-            selectedOffice.getLatitude(),
-            selectedOffice.getLongitude(),
-            entityList);
-
-    List<Entity> filteredListByDateRange;
     try {
-      filteredListByDateRange =
-          QueryHelper.filterOutOfDateRangeListings(
-              filteredEntityListByLatitude, startDate, endDate);
+      entityList =
+          runAllFiltersOnListings(
+              entityList, selectedOffice, distanceInKilometers, startDate, endDate);
     } catch (InvalidDateRangeException e) {
-      // If DateRange malformed
       response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
       return;
+    } catch (ApiException | InterruptedException | IOException e) {
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      return;
     }
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(filteredListByDateRange));
+    response.getWriter().println(gson.toJson(entityList));
+  }
+
+  private List<Entity> runAllFiltersOnListings(
+      List<Entity> originalEntities,
+      Office selectedOffice,
+      double distanceInKilometers,
+      Instant startDate,
+      Instant endDate)
+      throws InvalidDateRangeException, ApiException, InterruptedException, IOException {
+
+    List<Entity> filteredEntities =
+        CoordinateCalculator.filterOutOfRangeLatitudeEntities(
+            distanceInKilometers,
+            selectedOffice.getLatitude(),
+            selectedOffice.getLongitude(),
+            originalEntities);
+
+    filteredEntities =
+        QueryHelper.filterOutOfDateRangeListings(filteredEntities, startDate, endDate);
+
+    int distanceFromOfficeKilometers = 2;
```

Comment:
use `private static final DISTANCE_FROM_OFFICE_KM = 2;` instead for constants

Reply:
Will do! Thank you!